sphinx/directives/__init__.py

```diff
@@ -316,16 +317,22 @@ def run(self) -> list[Node]:
             docutils.unregister_role('')
             return []
         role_name = self.arguments[0]
-        role, messages = roles.role(role_name, self.state_machine.language,
+        messages: list[SystemMessage | system_message] = []
+        messages_: list[SystemMessage]
```

Comment:
``SystemMessage`` is a Docutils exception type, only raised on warnings with severity greater than ``halt_level`` -- it shouldn't be returned as a node type?

Reply:
This is what the new docutils types are saying. I can check upstream.

Reply:
It looks like the following should be ``nodes.system_message`` instead:

https://github.com/python/typeshed/blob/57c524d8225faae98611e6b643bb29769afdb3fa/stubs/docutils/docutils/parsers/rst/roles.pyi#L21-L23

https://github.com/python/typeshed/blob/57c524d8225faae98611e6b643bb29769afdb3fa/stubs/docutils/docutils/parsers/rst/directives/__init__.pyi#L12-L14

That has been wrong for two years, but perhaps something in @danieleades' latest change made MyPy look at those functions?

A

Reply:
Yep, came to the same conclusion. will send update to typeshed.

Reply:
See https://github.com/python/typeshed/pull/12314